aea/configurations/validation.py

```diff
@@ -295,12 +297,26 @@ def validate_data_with_pattern(
         excludes_: List[Tuple[str]] = []
     else:
         excludes_ = excludes
-    pattern_path_value = {
+    original_config = {
         tuple(path): value for path, value in dict_to_path_value(pattern)
     }
-    data_path_value = {tuple(path): value for path, value in dict_to_path_value(data)}
+    overrides = {tuple(path): value for path, value in dict_to_path_value(data)}
     errors = []
 
+    # this is a workaround to fix the type of numeric keys as they can only be represented as strs in the json overrides
+    for path in original_config:
+        path_as_str = tuple(map(str, path))
+        if path_as_str in overrides and path not in overrides:
+            value = overrides[path_as_str]
+            del overrides[path_as_str]
+            up_to_last_key = data
+            for key in path_as_str[:-1]:
+                up_to_last_key = up_to_last_key[key]
+            del up_to_last_key[path_as_str[-1]]
+            overrides[path] = value
+            vals = reduce(lambda d, key: {key: d}, reversed(path), value)
+            update_nested_dict(data, vals)
```

Comment:
Whats this about?

Reply:
Checks the type of the original format in the `skill.yaml` and uses this for the override. The overrides can only contain `str` keys, but they might be `int`, `float`, `bool`, etc.